### DISCUSSION TITLE
Welcome to LiteXDbHelper Discussions!

### DISCUSSION THREAD

**Discussion by a-patel** [a-patel/LiteXDbHelper#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by danthanhtrung** [comment]
LiteXDbHelper.Demo nothing any code for this packet

**Comment by a-patel** [comment]
@danthanhtrung Thank you to bring this to my attention.

I will add a Demo project for DbHelper packages. At this moment I am occupied with other important stuff.

I have opened a new issue for the same: https://github.com/a-patel/LiteXDbHelper/issues/6

**Comment by danthanhtrung** [comment]
You didn't make any throw error exception, for example in the case connection is wrong and more

**Comment by a-patel** [comment]
It will throw an exception when the connection opens.

For example:

```cs
        public int ExecuteNonQuery(string cmdText, CommandType cmdType, params DbParameter[] dbParameters)
        {
            using (var connection = new SqlConnection(ConnectionString))
            {
                using (var command = CreateCommand(connection, cmdText, cmdType, dbParameters))
                {
                    connection.Open();  // connection opens
                    var result = command.ExecuteNonQuery();
                    connection.Close();
                    return result;
                }
            }
        }
```

**Reply by danthanhtrung** [reply]
Thank you !
I was wrong, I didn't see your source code on Github, so I implement Nuget - LiteX.DbHelper.SqlServer it should look like this, and everything works fine, I haven't tested perfomance yet.

```
public User Login(string userName, string passWord)
        {
            var _dbHelper = new SqlHelper(_appSettings.SqlServerConnection_IdDB);
            var spName = "usp_Authenticate";
            var paramName = DbHelper.CreateParameter("@UserName", userName);
            var paramPass = DbHelper.CreateParameter("@UserPassword", passWord);
            var user = _dbHelper.GetSingle<User>(spName, CommandType.StoredProcedure, paramName, paramPass);
            _dbHelper.Dispose();
            return user;
        }
```
